tests/cpp/tree/hist/test_evaluate_splits.cc

```diff
@@ -108,5 +105,142 @@ TEST(HistEvaluator, Apply) {
   ASSERT_EQ(tree.Stat(tree[0].LeftChild()).sum_hess, 0.6f);
   ASSERT_EQ(tree.Stat(tree[0].RightChild()).sum_hess, 0.7f);
 }
+
+TEST(HistEvaluator, CategoricalPartition) {
+  int static constexpr kRows = 128, kCols = 1;
+  using GradientSumT = double;
+  std::vector<FeatureType> ft(kCols, FeatureType::kCategorical);
+
+  TrainParam param;
+  param.UpdateAllowUnknown(Args{{"min_child_weight", "0"}, {"reg_lambda", "0"}});
+
+  size_t n_cats{8};
+
+  auto dmat =
+      RandomDataGenerator(kRows, kCols, 0).Seed(3).Type(ft).MaxCategory(n_cats).GenerateDMatrix();
```

Comment:
I really like how we can generate random data with a [fluent interface](https://en.wikipedia.org/wiki/Fluent_interface).